### PR TITLE
[global-collect] Layout, docs improvements and code refactors

### PR DESCRIPTION
**Pull request by jyotigautam**
* `credo` issues resolved 
* Fixed some layout issues
* Corrected `amount` in examples
* Ran the elixir 1.6 code formatter
  - Used sigils in mocks
* Removed unnecessary functions
  - Reduced arity of `add_money`
* Refactored Timex usage
* Removed a test on `validate_config` as it is already tested.

### CODE REVIEW

File: lib/gringotts/gateways/global_collect.ex [hunk 1]
```diff
@@ -433,7 +433,7 @@ defmodule Gringotts.Gateways.GlobalCollect do
     datetime = Timex.now |> Timex.local
     strftime_str = Timex.format!(datetime, "%a, %d %b %Y %H:%M:%S ", :strftime)
     time_zone = Timex.timezone(:local, datetime)
-    time = strftime_str <>"#{time_zone.abbreviation}"
+    time = strftime_str <> "#{time_zone.abbreviation}"
```
Comment (oyeb) [inline lib/gringotts/gateways/global_collect.ex]: `time` is unused and `strftime_str <> "#{time_zone.abbreviation}"` need not be bound to it.

File: lib/gringotts/gateways/global_collect.ex [hunk 2]
```diff
@@ -423,8 +423,8 @@ defmodule Gringotts.Gateways.GlobalCollect do
 
   defp create_headers(path, opts) do
     time = date
```
[the remark this answers is not in the document]
Comment (jyotigautam) [reply lib/gringotts/gateways/global_collect.ex]: @oyeb What's wrong in line 182?

File: lib/gringotts/gateways/global_collect.ex [hunk 3]
```diff
@@ -14,41 +14,42 @@ defmodule Gringotts.Gateways.GlobalCollect do
...
-  | `shippingAddress`        |  implemented    |
-  | `name`                   |  implemented    |
-  | `skipAuthentication`     |  implemented    |
+  | `merchantCustomerId`     |  Identifier for the consumer that can be used as a search criteria in the Global Collect Payment Console |
+  | `description`            |  Descriptive text that is used towards to consumer, either during an online checkout at a third party and/or on the statement of the consumer |
+  | `dob`                    |  The date of birth of the consumer Format: YYYYMMDD   |
+  | `company`                |  Name of company, as a consumer                       |
+  | `email`                  |  Email address of the consumer                        |
+  | `phone`                  |  Phone number of the consumer                         |
+  | `invoice`                |  Object containing additional invoice data            |
```
Comment (oyeb) [inline lib/gringotts/gateways/global_collect.ex]: The schema of this `Object` is not documented.

File: lib/gringotts/gateways/global_collect.ex [hunk 4]
```diff
@@ -443,19 +421,36 @@ defmodule Gringotts.Gateways.GlobalCollect do
 
   defp respond({:ok, %{status_code: code, body: body}}) when code in [200, 201] do
     case decode(body) do
-      {:ok, results} -> {:ok, Response.success(raw: results, status_code: code)}
+      {:ok, results} ->
+        {
+          :ok,
+          Response.success(
+            authorization: results["payment"]["id"],
```
Comment (oyeb) [inline lib/gringotts/gateways/global_collect.ex]: Please populate other fields of the `Response` struct too, like `avs_result`, `cvv_result`, `message` and`fraud_result`

**Comment by oyeb** [comment]
* It appears that `GlobalCollect` _requires_ that some optional params be set, but this has not been documented.
* Some optional params are _assumed_ to be present and their absence is not handled ie, a `null` is rendered to the request.
* The fact that GC works in cents is not documented.

File: lib/gringotts/gateways/global_collect.ex (hunk 3 above)
Comment (jyotigautam) [reply lib/gringotts/gateways/global_collect.ex]: Structure of invoice object is shown in the examples if you read further in this file along with billingAddress,  shippingAddress and name.In addition to this, I am adding a reference to the link showing the required fields structure in Globalcollect docs.

Comment (oyeb) [reply lib/gringotts/gateways/global_collect.ex]: I did notice it's presence in the examples, but an example cannot define the schema. The `invoice` example is:

Comment (jyotigautam) [reply lib/gringotts/gateways/global_collect.ex]: Ya, I agree that examples can't define the schema so, for more information, I have added a link to refer.

File: test/gateways/global_collect_test.exs [hunk 5]
```diff
@@ -69,11 +69,18 @@ defmodule Gringotts.Gateways.GlobalCollectTest do
...
+      api_key_id: "e5743abfc360ed12"
+    }
+  ]
+
   @options [
     config: %{
-      secret_api_key: "some_secret_api_key",
-      api_key_id: "some_api_key_id",
-      merchant_id: "some_merchant_id"
+      secret_api_key: "Qtg9v4Q0G13sLRNcClWhHnvN1kVYWDcy4w9rG8T86XU=",
```
Comment (oyeb) [inline test/gateways/global_collect_test.exs]: Please revert this change.

File: test/gateways/global_collect_test.exs [hunk 6]
```diff
@@ -90,6 +97,14 @@ defmodule Gringotts.Gateways.GlobalCollectTest do
     skipAuthentication: "true"
   ]
 
+  describe "validation arguments check" do
```
Comment (oyeb) [inline test/gateways/global_collect_test.exs]: You don't have to test this functionality, it's already tested.
